src/main/java/ch/njol/skript/lang/Loop.java

```diff
@@ -0,0 +1,34 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.lang;
+
+/**
+ * Represents a loop section.
+ * 
+ * @see ch.njol.skript.sections.SecWhile
+ * @see ch.njol.skript.sections.SecLoop
+ */
+public interface Loop extends Debuggable {
```

Comment:
Can you abstract more things into this? E.g. resetting/exiting a loop

Reply:
Maybe it should extends SyntaxElement instead of Debuggable

Reply:
> Maybe it should extends SyntaxElement instead of Debuggable

Why? also it extends Debuggable to use Debuggable#toString()

Reply:
Oh I thought SyntaxElement extended Debuggable, apparently it doesn't. 
The reason I think it should also extend SyntaxElement is because every loop is a syntax element (and for it's uses, also a section, but that is an abstract class and IMO we should make this an abstract class).